cdist/conf/type/__pacman_conf/manifest

```diff
@@ -0,0 +1,111 @@
+#get params
+section=$(cat "$__object/parameter/section")
+key=$(cat "$__object/parameter/key")
+value=$(cat "$__object/parameter/value")
+file=$(cat "$__object/parameter/file" 2>/dev/null || echo "")
+state=$(cat "$__object/parameter/state" 2>/dev/null || echo "present" )
+
+#path variable default /etc/pacman.d
+sec_path="/etc/pacman.d"
+
+#allowed keys (from man pacman.conf)
+allowed_option_keys=( "RootDir" "DBPath" "CacheDir" "GPGDir" "LogFile" "HoldPkg" "IgnorePkg" "IgnoreGroup" "Include" "Architecture" "XferCommand" "NoUpgrade" "NoExtract" "CleanMethod" "SigLevel" "LocalFileSigLevel" "RemoteFileSigLevel" )
```

Comment:
Can we use a plain variable here? Even though I know that arrays *might* be neat for using, a standard variable does the same job and is portable accross various operating systems (your type might be running on aix to configure archlinux)

example:

allowed_option_keys="RootDir DBPath"

ok=0
for key in $allowed_option_keys; do
  if [ "$key" = "$given_key" ]; then ok=1; break; fi
done

if [ "$ok" -ne 1 ]; then
 echo "Invalid key $given_key" >&2
 exit 1
fi